### PR TITLE
[UTOPIA-943]fix pia tab nav btn

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- the pia  page nav button behaviour fix 
- the next and back buttons will follow the same behavior, from disclosure to additional risk tab,  it will only show back in last tab and next in first tab

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIANavButton/index.tsx [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
-const PIANavButton = ({ pia }: INavButton) => {
+const PIANavButton = ({ pages }: INavButton) => {
   const { pathname } = useLocation();
   const navigate = useNavigate();
+
+  const handleNavBtn = (direction: string) => {
+    const currentTabObj = pages.filter((page) => page.link === pathname);
+    let resultTab: INavbarItem[] = [];
+    if (direction === 'next') {
+      resultTab = pages.filter((page) => page.id === currentTabObj[0].id + 1);
```
Comment (antsand) [inline src/frontend/src/components/public/PIANavButton/index.tsx:13]: I do feel this is a super dangerous way to do it. 
For one, it 'id' is not generated and it is manually written. Something I am not happy with. 
Two, the order is based on how the object is structured in the array not the id. 

A better way is to get the index of the object on 'line 10'. and then check if (index>0) index-- and if (index <pages.length-1) index++

File: src/frontend/src/components/public/PIANavButton/index.tsx [hunk 2]
```diff
@@ -1,42 +1,48 @@
...
           <div className="horizontal-divider"></div>
-          <div className="form-buttons">
-            <button
-              className="bcgovbtn bcgovbtn__secondary btn-back"
-              onClick={handleBack}
-            >
-              Back
-            </button>
+          <div className="form-buttons ">
+            {!pathname.includes('collectionUseAndDisclosure') && (
```
Comment (antsand) [inline src/frontend/src/components/public/PIANavButton/index.tsx:34]: This type of check is asking for trouble. Check based on the length of the array and the current index.

File: src/frontend/src/components/public/PIANavButton/index.tsx [hunk 3]
```diff
@@ -1,42 +1,48 @@
...
+              <button
+                className="bcgovbtn bcgovbtn__secondary btn-back"
+                onClick={handleBack}
+              >
+                Back
+              </button>
+            )}
             {!pathname.includes('additionalRisks') && (
```
Comment (antsand) [inline src/frontend/src/components/public/PIANavButton/index.tsx:42]: same comment as above. I remember mentioning this in the previous PR too.

File: src/frontend/src/components/public/PIANavButton/index.tsx (hunk 1 above)
Comment (binggaofw) [reply src/frontend/src/components/public/PIANavButton/index.tsx:13]: let me re-write it. You are right, this implementation does not very robust

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
Why this hacky? And I do not encourage introducing refactors just to suffice the current purpose. +4s and +1s aren't the right way.

File: src/frontend/src/components/public/PIANavButton/index.tsx [hunk 4]
```diff
@@ -1,50 +1,92 @@
...
-import { PiaFormSideNavPages } from '../PIASideNav/pia-form-sideNav-pages';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
 import { INavButton } from './interface';
 
-const PIANavButton = ({ pia }: INavButton) => {
+const PIANavButton = ({ pages, isIntakeSubmitted, isDelegate }: INavButton) => {
   const { pathname } = useLocation();
   const navigate = useNavigate();
+
+  const handleNavBtn = (direction: string) => {
```
Comment (antsand) [inline src/frontend/src/components/public/PIANavButton/index.tsx:8]: This function can be optimized. Can work on it later.

File: src/frontend/src/components/public/PIANavButton/index.tsx [hunk 5]
```diff
@@ -1,50 +1,92 @@
...
-                className="bcgovbtn  bcgovbtn__secondary btn-next"
-                onClick={handleNext}
-              >
-                Next
-              </button>
-            )}
-          </div>
-        </div>
-      )}
+      {!isDelegate
```
Comment (antsand) [inline src/frontend/src/components/public/PIANavButton/index.tsx:46]: This condition must come from the page->state>prev>condition. 
Why is there an if/else for this?

**Review by Bottle7 — DISMISSED** [review]
This is working well for me 🚀 Thanks, Bing!

File: src/frontend/src/components/public/PIANavButton/index.tsx (hunk 5 above)
Comment (binggaofw) [reply src/frontend/src/components/public/PIANavButton/index.tsx:46]: the next Step page show the button differently based on PI or Non-PI. if it is PI, we show them next and back, but if non-PI, it display back and view pia list btn

File: src/frontend/src/components/public/PIASideNav/pia-form-sideNav-pages.ts [hunk 6]
```diff
@@ -138,6 +229,12 @@ export const PiaFormSideNavPages = (
         },
       ),
       enable: showPostIntakeTabs,
+      state: {
+        prev: {
+          condition: true,
+          action: -1,
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIASideNav/pia-form-sideNav-pages.ts:235]: nit: how about adding +1s and -1s as default actions.

**Review by kushal-arora-fw — DISMISSED** [review]

**Review by kushal-arora-fw — DISMISSED** [review]
Looks good 👍

**Review by kushal-arora-fw — APPROVED** [review]
Look good!

**Pull request merged**